### DISCUSSION TITLE
Select folders & files separately

### DISCUSSION THREAD

**Discussion by tomoayan** [SebastianMC/obsidian-custom-sort#47] [Q&A] [answered]
**EDIT:** Want to select and sort folder and files differently.
e.g. < a-z (Folders), > a-z (Files)
             ^   Super Simplified Example.

**Comment by SebastianMC** [comment, marked as answer]
Hi @tomoayan 

the relevant fragment of sorting specification would be:

```yaml
 /folders
  < a-z
 /:files
  > a-z
```

It groups all folders and sorts them alphabetically and then, below folders, groups all files and sorts them in reversed alphabetical order.

In order to work, the above snippet has to be put in the `sorting-spec:` key in YAML frontmatter of relevant note. The simplest case would be the `sortspec` note anywhere in the vault with additional explicit specification of target folder to sort, for clarity.

The full example to put in `sortspec` note would be then:

```yaml
---
sorting-spec: |
 target-folder: /
 /folders
  < a-z
 /:files
  > a-z
---
```

in result it would apply the sorting order to the root folder of the vault (and only, no subfolders affected).

> REMARK: indentation matters, and in this specific example, the order specifications ` < a-z` and ` > a-z` are nested by one more space relatively to the preceding lines `/folders` and `/:files`. It tells the sorting engine to apply the sorting rules to the preceding group of items. Otherwise (no nesting) the order would be applied to all items in the folder. And if two not-nested order specifications would occur, a _syntax error_ would be raised telling about duplicate sort order specification for the target folder.

Please let me know if you want to exercise variants of the above sorting specification, e.g.: to sort some other folder(s), not the root folder only.

**Comment by tomoayan** [comment]
> It groups all folders and sorts them alphabetically and then, below folders, groups all files and sorts them in reversed alphabetical order.

Thanks 👍
